compiler-core/src/config.rs

```diff
@@ -103,6 +104,13 @@ impl PackageConfig {
         }
     }
 
+    pub fn dependency_versions_for(&self, mode: Mode) -> Result<HashMap<String, Range>> {
```

Comment:
Where does this function get used? I couldn't find it.

I ask because I was very surprised that `requirement.version_range` reads from the filesystem (and presumably will have to call `git clone` for git deps), the IO seemed out of place. I wanted to learn the context in which it was used so that I could understand better.

Reply:
Good catch, it doesn't! Remanent of early work when I thought I might need to pre-check all the versions before actually loading the packages.